libraries/apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/gql.kt

```diff
@@ -726,22 +744,24 @@ class GQLSchemaExtension(
         directives.join(writer)
         write(" ")
       }
-      write("{\n")
-      indent()
-      operationTypesDefinition.join(writer, separator = "")
-      unindent()
-      write("}\n")
+      if (operationTypeDefinitions.isNotEmpty()) {
+        write("{\n")
+        indent()
+        operationTypeDefinitions.join(writer, separator = "")
+        unindent()
+        write("}\n")
+      }
```

Comment:
It is valid for a `GQLSchemaExtension` to not have any `operationTypeDefinitions`, for an example to add a directive to the schema:

```graphql
extend schema @foo
```

But it is not valid to have empty braces (`{}` ) in the grammar. Same with other type extensions below